app/controllers/application_controller.rb

```diff
@@ -2,4 +2,19 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :ensure_current_user, except: [:index, :show]
+
+  helper_method :current_user, :ensure_current_user
```

Comment:
ensure_current_user doesn't need to be a helper method - you don't use it in views.